### PR TITLE
EVG-16660 Add upstreamProject field to versions and mainline commits

### PR DESCRIPTION
**Pull request by khelif96**
[EVG-16660](https://jira.mongodb.org/browse/EVG-16660)

### Description 
Upstream Projects triggered versions will now link to the triggering task or version

Since we do not have the concept of builds on the spruce UI we will default to linking the version associated with a build.

### Screenshots

image

image

### Testing 
Staging and locally

### Evergreen PR 
https://github.com/evergreen-ci/evergreen/pull/5565

### CODE REVIEW

**Review by sophstad — APPROVED** [review]

File: src/pages/version/Metadata.tsx [hunk 1]
```diff
@@ -92,6 +101,20 @@ export const Metadata: React.VFC<Props> = ({ loading, version }) => {
...
       {manifest && <ManifestBlob manifest={manifest} />}
+      {upstreamProject && (
+        <P2>
+          Triggered from:{" "}
+          <StyledRouterLink
+            to={
+              triggerType === ProjectTriggerLevel.TASK
+                ? getTaskRoute(upstreamTask?.id)
+                : getVersionRoute(upstreamVersion?.id)
+            }
```
Comment (minnakt) [inline src/pages/version/Metadata.tsx:112]: sorry I missed this one! I think optional chaining also may not be needed here

**Review by minnakt — APPROVED** [review]
😎🔥

**Comment by SupaJoon** [comment]
These code changes add the new label to folded commits in the history table but not the inactive commit labels in the commit chart. Is that intentional?

**Comment by khelif96** [comment]
image
image
Done

> These code changes add the new label to folded commits in the history table but not the inactive commit labels in the commit chart. Is that intentional?

**Review by SupaJoon — APPROVED** [review]

**Pull request merged**